internal/cachecontroller/cache_controller.go

```diff
@@ -100,15 +103,29 @@ func (c *InMemoryCacheController) DetermineInvalidation(
 		return entry.LastModified
 	}
 
-	lastModified, err, _ := c.sf.Do(storeID, func() (interface{}, error) {
-		return c.findChangesAndInvalidate(ctx, storeID)
-	})
-
-	if err != nil {
-		return time.Time{}
+	c.mu.Lock()
+	_, present := c.inflightInvalidations[storeID]
+	if !present {
+		c.inflightInvalidations[storeID] = struct{}{}
 	}
+	c.mu.Unlock()
+	if !present {
+		span.SetAttributes(attribute.Bool("checkInvalidation", true))
+		// if the cache cannot be found, we want to invalidate entries in the background
+		// so that it does not block the answer path.
+		go func() {
+			// important to propagate the cancel without cancel to avoid
```

Comment:
```suggestion
			// important to propagate the context without cancel to avoid
```

Reply:
I had no idea `WithoutCancel` was a thing

Reply:
Yeah...this is something I learn today - we should look into using it in some of the check / graph code because it will give us ability to link to the original span (which we will lack if we use a brand new context)